### ISSUE TITLE
False positives for recursions

### ISSUE THREAD

**Issue by neriusmika** [mobility-university/flake8-newspaper-style#2]
There are warnings when methods are called recursively.

**Comment by lindt** [comment]
could you post an example where it fails?

**Comment by neriusmika** [comment]
Just any recursive method call. For example
```python
class Test:
    counter = 0

    def do_something(self):
        if self.counter < 1:
            self.counter += 1
            self.do_something()
```

**Comment by lindt** [comment]
I added this example in #7 as unittest. No false positive in this case.
Could you maybe retry? Maybe it got fixed in some versions in between already.
Or there is another recursion example, where it still produces false positives.
Thx!!

**Comment by neriusmika** [comment]
I installed newspaper style to find these errors, but now this plugin breaks other plugins and I have many false positives in other places.
![image](https://user-images.githubusercontent.com/23192030/194411010-d7bfe6d2-f26f-4c0b-9624-394671b8510d.png)
After disabling newspaper style, there are no errors at all.

Latest flake8 5, with the latest plugins.

**Comment by lindt** [comment]
thx. nice find.
haven't thought, that removing the decorators in this plugin has effects on other plugins leading to false positives.

just released v0.4.1. This solves it.
Would be awesome if you could test it. Thx!!

**Comment by neriusmika** [comment]
Still have the same errors with version 0.4.1.

**Comment by lindt** [comment]
hmm....i was able to reproduce your errors before, now cannot.

guess then we need a test case for it.
not sure how to start. would be nice if could minimize your test case.

**Comment by neriusmika** [comment]
I'll try to investigate where is the exact problem. The project I'm checking is big, and I can't just copy everything. It will take a while. Thank you.

**Comment by neriusmika** [comment]
The problem is in the `filter_function_calls_for_class_variables` function. The ast tree is modified there, and other plugins get an incomplete tree. The simplest solution is to first copy the ast tree with deepcopy (I quickly checked, the problem disappears). I did not analyze whether it is possible to do without modifying the tree at all.

**Comment by neriusmika** [comment]
I checked the same code sample with version 0.4.1, and it gives me an error
![image](https://user-images.githubusercontent.com/23192030/194463506-76107159-d25e-4867-9a83-e3fd8c2262e1.png)

By the way, deepcopy sometimes throws python RecursionError, it's better to avoid node modification.

**Comment by lindt** [comment]
thx. I guess, not I understood your case.
recursion should now work.
mutation of AST also is avoided now.

if you could try out v0.4.2, would be nice. Thx!

**Comment by neriusmika** [comment]
I checked several large projects, all problems disappeared. Thank you!
